include/osquery/expected.h

```diff
@@ -104,7 +104,14 @@ class Expected final {
   Expected(const Expected&) = delete;
   Expected(ErrorBase* error) = delete;
 
-  Expected& operator=(Expected&& other) = default;
+  Expected& operator=(Expected&& other) {
+    if (this != &other) {
+      object_ = std::move(other.object_);
+      other.errorChecked_ = true;
```

Comment:
I think you need to reassign the value of `errorChecked_` too.
```c++
  errorChecked_ = other.errorChecked_;
```

And may be it is a good idea to check the value of `errorChecked_` before reassigning it.